bestie/server/service.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// List of event ID names defined by build_event_stream.proto.
+// These formal event names are mapped to Prometheus metric label names.
+var besEventIds = map[string]string{
+	"Unknown":                       "unknown", // should not normally see any of these
+	"Progress":                      "progress",
+	"Started":                       "started",
+	"CommandLine":                   "unstructured_command_line", // CommandLine replaced by UnstructuredCommandLine
+	"UnstructuredCommandLine":       "unstructured_command_line",
+	"StructuredCommandLine":         "structured_command_line",
+	"WorkspaceStatus":               "workspace_status",
+	"OptionsParsed":                 "options_parsed",
+	"Fetch":                         "fetch",
+	"Configuration":                 "configuration",
+	"TargetConfigured":              "target_configured",
+	"Pattern":                       "pattern",
+	"PatternSkipped":                "pattern_skipped",
+	"NamedSet":                      "named_set",
+	"TargetCompleted":               "target_completed",
+	"ActionCompleted":               "action_completed",
+	"UnconfiguredLabel":             "unconfigured_label",
+	"ConfiguredLabel":               "configured_label",
+	"TestResult":                    "test_result",
+	"TestSummary":                   "test_summary",
+	"TargetSummary":                 "target_summary",
+	"BuildFinished":                 "build_finished",
+	"BuildToolLogs":                 "build_tool_logs",
+	"BuildMetrics":                  "build_metrics",
+	"Workspace":                     "workspace",
+	"BuildMetadata":                 "build_metadata",
+	"ConvenienceSymlinksIdentified": "convenience_symlinks_identified",
+}
+
+// BES Prometheus metric definition.
+// Use this for untagged counters.
+type besPromMetric struct {
+	label string
+	ctr   prometheus.Counter
+}
+
+// BES Prometheus metric vector definition.
+// Use this for tagged counters.
+type besPromMetricVec struct {
+	ctr *prometheus.CounterVec
+}
+
+// BES Prometheus histogram definition.
+type besPromHistogram struct {
+	ctr prometheus.Histogram
+}
+
+// Number of BES messages seen by event type.const
+type besEvents struct {
+	bazelBuildsTotal besPromMetric
+	bazelEventsTotal besPromMetricVec
+}
+
```

Comment:
This file is intentionally structured this way to allow for the easy addition of BigQuery service metrics in the PR that's coming for INFRA-423.